src/comms/CommsManager.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2025 The MMapper Authors
+
+#include <QString>
+#include <QObject>
+#include <QSet>
+#include <QPair>
+
+#include "../global/utils.h"
+
+class GmcpMessage;
+
+enum class NODISCARD CommType {
+    TELL,
+    WHISPER,
+    GROUP,
+    SAY,
+    EMOTE,
+    NARRATE,
+    YELL,
+    PRAY,
+    SHOUT,
+    SING,
+    ASK,
+    SOCIAL
+};
+
+enum class NODISCARD CommCategory {
+    DIRECT,  // tells, whispers
+    LOCAL,   // say, emote, social
+    GLOBAL   // narrate, yell, pray, shout, sing, ask (questions)
+};
+
+enum class NODISCARD TalkerType {
+    YOU,      // Messages sent by the player (talker: "you")
+    PLAYER,   // Regular player (no talker-type specified)
+    NPC,      // NPC (talker-type: "npc")
+    ALLY,     // Ally (talker-type: "ally")
+    NEUTRAL,  // Neutral (talker-type: "neutral")
+    ENEMY     // Enemy (talker-type: "enemy")
+};
+
+struct NODISCARD CommMessage final
+{
+    CommType type = CommType::SAY;
+    CommCategory category = CommCategory::LOCAL;
+    QString sender;
+    QString message;
+    QString timestamp;
+    TalkerType talkerType = TalkerType::PLAYER;
+};
+
+class CommsManager final : public QObject
+{
+    Q_OBJECT
+
+public:
+    explicit CommsManager(QObject *parent);
+    ~CommsManager() override;
+
+    DELETE_CTORS_AND_ASSIGN_OPS(CommsManager);
+
+public slots:
+    void slot_parseGmcpInput(const GmcpMessage &msg);
+    void slot_parseRawGameText(const QString &rawText);
```

Comment:
I don't see where these signals and slots are connected. I guess it got missed in the PR split?